nibabel/cifti2/tests/test_cifti2.py

```diff
@@ -0,0 +1,361 @@
+    )
+
+    vs[0] = 10
+    assert_equal(vs[0], 10)
+    assert_equal(len(vs), 3)
+    vs = ci.Cifti2Vertices(vertices=[0, 1, 2])
+    assert_equal(len(vs), 3)
+
+
+def test_cifti2_transformationmatrixvoxelindicesijktoxyz():
+    tr = ci.Cifti2TransformationMatrixVoxelIndicesIJKtoXYZ()
+    assert_raises(ci.Cifti2HeaderError, tr.to_xml)
+
+
+def test_cifti2_surface():
+    s = ci.Cifti2Surface()
+    assert_raises(ci.Cifti2HeaderError, s.to_xml)
+
+
+def test_cifti2_volume():
+    vo = ci.Cifti2Volume()
+    assert_raises(ci.Cifti2HeaderError, vo.to_xml)
+
+
+def test_cifti2_vertexindices():
+    vi = ci.Cifti2VertexIndices()
+    assert_equal(len(vi), 0)
+    assert_raises(ci.Cifti2HeaderError, vi.to_xml)
+    vi.extend(np.array([0, 1, 2]))
+    assert_equal(len(vi), 3)
+    assert_equal(
+        vi.to_xml().decode('utf-8'),
+        '<VertexIndices>0 1 2</VertexIndices>'
+    )
+    assert_raises(ValueError, vi.__setitem__, 0, 'a')
+    vi[0] = 10
+    assert_equal(vi[0], 10)
+    assert_equal(len(vi), 3)
+
+
+def test_cifti2_voxelindicesijk():
+    vi = ci.Cifti2VoxelIndicesIJK()
+    assert_raises(ci.Cifti2HeaderError, vi.to_xml)
+
+    vi = ci.Cifti2VoxelIndicesIJK()
+    assert_equal(len(vi), 0)
+    assert_raises(ci.Cifti2HeaderError, vi.to_xml)
+    vi.extend(np.array([[0, 1, 2]]))
+    assert_equal(len(vi), 1)
+    assert_equal(vi[0], [0, 1, 2])
+    vi.append([3, 4, 5])
+    assert_equal(len(vi), 2)
+    vi.append([6, 7, 8])
+    assert_equal(len(vi), 3)
+    del vi[-1]
+    assert_equal(len(vi), 2)
+
+    assert_equal(vi[1], [3, 4, 5])
+    vi[1] = [3, 4, 6]
+    assert_equal(vi[1], [3, 4, 6])
+    assert_raises(ValueError, vi.__setitem__, 'a', [1, 2, 3])
+    assert_raises(TypeError, vi.__setitem__, [1, 2], [1, 2, 3])
+    assert_raises(ValueError, vi.__setitem__, 1, [2, 3])
+    assert_equal(vi[1, 1], 4)
+    assert_raises(ValueError, vi.__setitem__, [1, 1], 'a')
+    assert_equal(vi[0, 1:], [1, 2])
+    vi[0, 1] = 10
+    assert_equal(vi[0, 1], 10)
+    vi[0, 1] = 1
+
+    #test for vi[:, 0] and other slices
+    assert_raises(NotImplementedError, vi.__getitem__, (slice(None), 0))
+    assert_raises(NotImplementedError, vi.__setitem__, (slice(None), 0), 0)
+    assert_raises(NotImplementedError, vi.__delitem__, (slice(None), 0))
+    assert_raises(ValueError, vi.__getitem__, (0, 0, 0))
+    assert_raises(ValueError, vi.__setitem__, (0, 0, 0), 0)
+
+    assert_equal(
+        vi.to_xml().decode('utf-8'),
+        '<VoxelIndicesIJK>0 1 2\n3 4 6</VoxelIndicesIJK>'
+    )
+    assert_raises(TypeError, ci.Cifti2VoxelIndicesIJK, [0, 1])
+    vi = ci.Cifti2VoxelIndicesIJK([[1, 2, 3]])
+    assert_equal(len(vi), 1)
+
+
+def test_matrixindicesmap():
+    mim = ci.Cifti2MatrixIndicesMap(0, 'CIFTI_INDEX_TYPE_LABELS')
+    volume = ci.Cifti2Volume()
+    volume2 = ci.Cifti2Volume()
+    parcel = ci.Cifti2Parcel()
+
+    assert_is_none(mim.volume)
+    mim.append(volume)
+    mim.append(parcel)
```

Comment:
This appears to have declared the index map as LABELS type, which cannot accept volume or parcel elements.  Was it supposed to be PARCELS type?

Taking a look at the standard document, it appears that it wasn't stated that a volume or parcel element should not exist in a LABELS type mapping (or similar for other types - basically, if it wasn't mentioned in the lists under the diagram, it shouldn't exist).  I'll put it on my list of oversights to correct for the next document.